app/services/post_parser.py

```diff
@@ -28,3 +28,44 @@ def extract_contact_info(text):
     matches_email.extend(re.findall(email_pattern, text))
     contact = {"mobile": matches_phone, "email": matches_email}
     return contact
+
+
+def seperate_posts_from_whatsapp_bulk_text(bulk):
+    """
+    Add WhatApp text bulk.
+
+    Parameters
+    ----------
+    bulk: string
+
+    Returns
+    -------
+    list (of strings) containing each message as an element excluding
+    repeated format
+
+    Example
+    -------
+    ['message 1',
+    'message 2',
+    'message 3']
+    """
+    # These patterns extract the message right after the repeated format exclusively
+    unsaved_ctc_sep_pat = (
+        r"\[.*?\] \+\d+ \d{2} \d{7}:\s*(.*)"  # Unsaved Contact Seperator Pattern
+    )
+
+    saved_ctc_sep_pat = r"\[.*?\] .*?, .*?:\s*(.*)"  # Saved Contact Seperator Pattern
```

Comment:
Saved names are of free text value, not necessarily with comma, but it's necessary to have `:` to end.
We can test also on:
```
[8:17 pm, 17/06/2024] TheKing : وظيفة أخرى 01212345678 [8:17 pm, 17/06/2124]
[8:17 pm, 17/06/2024] Another Name : وظيفة جديدة 01212349678
```
Output should be:
```
1. TheKing : وظيفة أخرى 01212345678 [8:17 pm, 17/06/2124]
2. Another Name : وظيفة جديدة 01212349678
```